formats/opentype.fathom

```diff
@@ -88,24 +88,24 @@ def table_directory (file_start : Pos) = {
             fun (table_id : Repr tag) =>
             fun (table_format : Format) => {
                 // TODO: let formats
-                table_record <- unwrap (Repr table_record) (find_table _ table_records table_id),
+                table_record <- unwrap (find_table table_records table_id),
                 link <- link_table file_start table_record table_format,
             };
 
         let required_table_with_len =
             fun (table_id : Repr tag) =>
             fun (table_format : (U32 -> Format)) => {
                 // TODO: let formats
-                table_record <- unwrap (Repr table_record) (find_table _ table_records table_id),
+                table_record <- unwrap (find_table table_records table_id),
                 link <- link_table file_start table_record (table_format table_record.length),
             };
 
         let optional_table =
             fun (table_id : Repr tag) =>
             fun (table_format : Format) =>
-                option_fold (Repr table_record) Format {}
+                option_fold ({} : Format)
```

Comment:
It would be nice to postpone the elaboration of `{}` and `()` until later. At the moment we eagerly synthesise them to record literals.